exchange/src/exchange/traders/trader_types/generic_trader.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "trader_type.hpp"
+
+#include <string>
+#include <unordered_map>
+
+namespace nutc {
+namespace manager {
+
+class GenericTrader {
+public:
+    std::string
+    get_id() const
+    {
+        return USER_ID;
+    }
+
+    virtual TraderType get_type() const = 0;
+
+    double
+    get_capital() const
+    {
+        return capital_;
+    }
+
+    bool
+    is_active() const
+    {
+        return is_active_;
+    }
+
+    void
+    set_active(bool active)
+    {
+        is_active_ = active;
+    }
```

Comment:
this is c++ not java

```suggestion
    bool
    active() const noexcept
    {
        return is_active_;
    }

    bool&
    active() noexcept
    {
        return is_active;
    }
```

Reply:
See here it’s a bool but we use a reference to set it